### PR TITLE
Run browserstack tests with an allowance for flakiness

### PR DESCRIPTION
**Pull request by mattsoulanille**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4691)

### CODE REVIEW

File: tfjs-automl/package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "test": "yarn && yarn build && karma start",
     "test-dev": "karma start",
     "test-node": "tsc && ts-node --transpile-only --skip-ignore --project tsconfig.test.json src/test_node.ts",
+    "run-flaky": "node ../scripts/run_flaky.js",
```
Comment (mattsoulanille) [inline tfjs-automl/package.json:16]: This is necessary because `npm-run-all` can only run yarn scripts.

**Comment by mattsoulanille** [comment]
[Here's a nightly run](https://console.cloud.google.com/cloud-build/builds/304c4f6a-fae6-41bb-9b05-8087dc6779e0?project=834911136599) since this changes nightly logic as well. Not yet complete at the time of writing.

**Comment by mattsoulanille** [comment]
> [Here's a nightly run](https://console.cloud.google.com/cloud-build/builds/304c4f6a-fae6-41bb-9b05-8087dc6779e0?project=834911136599) since this changes nightly logic as well. Not yet complete at the time of writing.

Failed due to flaky (?) webdriver test. [Here's another nightly run](https://console.cloud.google.com/cloud-build/builds/afe5be6a-3141-4b3c-ac25-3cad85273180?project=834911136599).

**Review by pyu10055 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4691)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @lina128 and @mattsoulanille)

___
*[tfjs-backend-wasm/scripts/test-ci.sh, line 16 at r3](https://reviewable.io/reviews/tensorflow/tfjs/4691#-MTgza-LEkASh8B-UWVi:-MTgza-M4qEluW44mY_t:b-3eer7j) ([raw file](https://github.com/tensorflow/tfjs/blob/79a757815fe3440778620ac85eaa6f0a8b49f9cf/tfjs-backend-wasm/scripts/test-ci.sh#L16)):*
> ```Shell
> 
> if [ "$NIGHTLY" = true ]; then
>   node ../scripts/run_flaky.js "yarn run-browserstack --browsers=bs_safari_mac"
> ```

should this use yarn target (yarn run-flaky) similar to other packages? similar for webgl, wasm and core.

**Review by pyu10055 — COMMENTED** [review]
Thanks!

> Reviewed 7 of 8 files at r1, 1 of 1 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4691#-:-MTh-ENrAOegJCJ_3ZLf:b-ucfknh)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @lina128 and @mattsoulanille)

**Review by mattsoulanille — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4691)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @lina128 and @pyu10055)

___
*[tfjs-backend-wasm/scripts/test-ci.sh, line 16 at r3](https://reviewable.io/reviews/tensorflow/tfjs/4691#-MTgza-LEkASh8B-UWVi:-MTh-lJO5zI5YzMDAKdK:b-nm478r) ([raw file](https://github.com/tensorflow/tfjs/blob/79a757815fe3440778620ac85eaa6f0a8b49f9cf/tfjs-backend-wasm/scripts/test-ci.sh#L16)):*
<details><summary><i>Previously, pyu10055 (Ping Yu) wrote…</i></summary><blockquote>

should this use yarn target (yarn run-flaky) similar to other packages? similar for webgl, wasm and core.
</blockquote></details>

It doesn't have to, but it may be a good idea. I wrote a `run-flaky` yarn rule for `tfjs-automl` and `tfjs-vis` because `npm-run-all` can only run yarn scripts and couldn't run `node ../scripts/run_flaky.js`. I could add a `run-flaky` rule to each other package if you want.

**Comment by mattsoulanille** [comment]
Actually, I think it's probably fine to leave `node ../scripts/run_flaky.js` as is, at least for now. Since we're in a monorepo, it should be fine to reference scripts outside the package. I can make another PR if we want to change this in the future.

**Pull request merged**